R/MsBackendCached.R

```diff
@@ -0,0 +1,614 @@
+setClassUnion("characterOrInteger", c("character", "integer"))
+
+#' @title Base MsBackend class providing data caching mechanism
+#'
+#' @aliases MsBackendCached-class MsBackendCached
+#'
+#' @description
+#'
+#' The `MsBackendCached` class is a rudimentary implementation of the
+#' [MsBackend] providing a simple mechanism to cache spectra data locally. This
+#' class is thought to be used as a base class for other `MsBackend`
+#' implementations to reuse its caching mechanism and avoid having to
+#' re-implement commonly used methods. This class is thus not thought to be
+#' used directly by a user.
+#'
+#' The provided caching mechanism by this class allows `MsBackend` instances
+#' to add or replace spectra variables even if the backend used by them does
+#' not allow to alter values (e.g. if a SQL database is used as a backend). Any
+#' replacement operation with `$<-` will add the specified values to a local
+#' `data.frame` within the `MsBackendCached` class that allows to *cache* these
+#' values (increasing obviously the memory demand of the object).
+#'
+#' Any data accessor functions of the extending `MsBackend` class (such as `$`
+#' or `msLevel` or `spectraData`) should first use `callNextMethod` to call the
+#' respective accessor of `MsBackendCached` that will evaluate if the
+#' requested spectra variable(s) are in the local cache and return these. If
+#' the requested spectra variables are not in the local cache, are also not in
```

Comment:
If the requested spectra variables are neither in the local cache, nor listed in the `@spectraVariables` slot

Reply:
Thanks